joint_trajectory_controller/test/test_trajectory.cpp

```diff
@@ -63,101 +75,80 @@ TEST_F(TestTrajectory, sample_trajectory) {
 
   trajectory_msgs::msg::JointTrajectoryPoint p1;
   p1.positions.push_back(1.0f);
-  builtin_interfaces::msg::Duration d1;
-  d1.sec = 1;
-  d1.nanosec = 0;
-  p1.time_from_start = d1;
-
-  trajectory_msgs::msg::JointTrajectoryPoint p2;
-  p2.positions.push_back(2.0f);
-  builtin_interfaces::msg::Duration d2;
-  d2.sec = 2;
-  d2.nanosec = 0;
-  p2.time_from_start = d2;
-
-  trajectory_msgs::msg::JointTrajectoryPoint p3;
-  p3.positions.push_back(3.0f);
-  builtin_interfaces::msg::Duration d3;
-  d3.sec = 3;
-  d3.nanosec = 0;
-  p3.time_from_start = d3;
-
-  full_msg->points.push_back(p1);
-  full_msg->points.push_back(p2);
-  full_msg->points.push_back(p3);
-
-  auto traj = joint_trajectory_controller::Trajectory(full_msg);
-
-  auto sample_p1 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p1);
-  EXPECT_EQ(1.0f, sample_p1->positions[0]);
-
-  auto sample_p11 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p11);
-  EXPECT_EQ(1.0f, sample_p11->positions[0]);
-
-  std::this_thread::sleep_for(1s);
-
-  auto sample_p2 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p1);
-  EXPECT_EQ(2.0f, sample_p2->positions[0]);
-
-  auto sample_p22 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p22);
-  EXPECT_EQ(2.0f, sample_p22->positions[0]);
-
-  std::this_thread::sleep_for(1s);
-
-  auto sample_p3 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p1);
-  EXPECT_EQ(3.0f, sample_p3->positions[0]);
-
-  auto sample_p33 = traj.sample(rclcpp::Clock().now());
-  ASSERT_NE(traj.end(), sample_p33);
-  EXPECT_EQ(3.0f, sample_p33->positions[0]);
-
-  std::this_thread::sleep_for(1s);
-
-  auto sample_end = traj.sample(rclcpp::Clock().now());
-  EXPECT_EQ(traj.end(), sample_end);
-
-  auto sample_end_end = traj.sample(rclcpp::Clock().now());
-  EXPECT_EQ(traj.end(), sample_end_end);
-}
-
-TEST_F(TestTrajectory, future_sample_trajectory) {
-  auto full_msg = std::make_shared<trajectory_msgs::msg::JointTrajectory>();
-  full_msg->header.stamp = rclcpp::Clock().now();
-  full_msg->header.stamp.sec += 2;  // extra padding
-
-  trajectory_msgs::msg::JointTrajectoryPoint p1;
-  p1.positions.push_back(1.0f);
-  builtin_interfaces::msg::Duration d1;
-  d1.sec = 1;
-  d1.nanosec = 0;
-  p1.time_from_start = d1;
+  p1.velocities.push_back(0.0f);
```

Comment:
Let's deal with cargo cult here and remove the `f`. There's plenty of other parts where this could be applied to but we are not in the 90s. Let's allow the compiler to figure this one out.

Reply:
removed all f's ccc6897